File: 648-replace-words/648-replace-words.cpp
class Solution {
public:
    string replaceWords(vector<string>& dic, string sentence) 
    {
        string str="", s="";
        sentence+=" ";
        sort(dic.begin(), dic.end(), []
            (const string& first, const string& second)
             {
                return first.size() < second.size();
             }
        );        
        for(int i=0; i<sentence.length(); i++)
        {
            if(sentence[i]!=' ')
                str += sentence[i];
            else
            {
                for(int j=0; j<dic.size(); j++)
                    if(dic[j]==str.substr(0, dic[j].length()))
                        str = dic[j];
                s = s + str + " ";
                str = "";
            }
        }
        s = s.substr(0,s.length()-1);
        return s;
    }
};

File: 648-replace-words/648-replace-words.java
class Solution {
    public String replaceWords(List<String> dictionary, String sentence) 
    {
        Collections.sort(dictionary, Collections.reverseOrder());
        String []sen = sentence.split("\\s+");
        String s = "", str="";
        for(int i=0; i<sen.length; i++)
        {
            str = sen[i];
            for(String dict:dictionary)
                if(dict.length() <= sen[i].length())
                    if(sen[i].substring(0, dict.length()).equals(dict))
                        str = dict;
            s += str + " ";
        }
        return s.substring(0, s.length()-1);
    }
}
